src/schema_compile_node.c

```diff
@@ -3311,7 +3312,20 @@ lys_compile_node_choice_child(struct lysc_ctx *ctx, struct lysp_node *child_p, s
 
 free_fake_node:
         /* free the fake parsed node and correct pointers back */
+
         cs_p->child = NULL;
+        if (ctx->ctx->flags & LY_CTX_SET_PRIV_PARSED) {
+            /* compiled node cannot point to the implicit case node */
+            LY_LIST_FOR((struct lysc_node *)((struct lysc_node_choice *)node)->cases, compiled_case) {
```

Comment:
A loop to find the last element is redundant, just take the first child `prev` pointer and you have it.

Reply:
ok, great